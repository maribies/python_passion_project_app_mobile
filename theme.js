export const theme = {
  margin: "20px",
  fonts: {
    xLarge: "33px",
    large: "26px",
    medium: "20px",
    default: "16px",
    small: "12.5px",
    family: "Georgia",
    weight: "normal",
  },
};
