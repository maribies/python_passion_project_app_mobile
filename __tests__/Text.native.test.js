import React from "react";
import { create } from "react-test-renderer";
import { ThemeProvider } from "emotion-theming";
import { H1, H2, H3, MainText, Caption } from "../Components/Text.native";
import { theme } from "../theme";

// This test is borderline unnecessary now that it is just rendering UI,
// But I'm keeping for now as it helped ensure that I passed the props correctly
// and extended the styles correctly using a new lib. So although it's not so
// helpful as a test going forward, it was helpful in the development and refactor
// of these elements.

test("correctly renders the text elements", () => {
  let text = create(
    <ThemeProvider theme={theme}>
      <MainText>Text of the main persuasion.</MainText>
    </ThemeProvider>
  );
  let title = create(
    <ThemeProvider theme={theme}>
      <H1>Title!</H1>
    </ThemeProvider>
  );
  let subtitle = create(
    <ThemeProvider theme={theme}>
      <H2>I am a subtitle.</H2>
    </ThemeProvider>
  );
  let heading = create(
    <ThemeProvider theme={theme}>
      <H3>Not quite a subtitle, but still more important than the rest.</H3>
    </ThemeProvider>
  );
  let caption = create(
    <ThemeProvider theme={theme}>
      <Caption>I am a Caption!</Caption>
    </ThemeProvider>
  );

  expect(text.toJSON()).toMatchSnapshot();
  expect(title.toJSON()).toMatchSnapshot();
  expect(subtitle.toJSON()).toMatchSnapshot();
  expect(heading.toJSON()).toMatchSnapshot();
  expect(caption.toJSON()).toMatchSnapshot();
});
