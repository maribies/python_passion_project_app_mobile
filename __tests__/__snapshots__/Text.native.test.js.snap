// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`correctly renders the text elements 1`] = `
<Text
  style={
    Object {
      "fontFamily": "FanwoodText_400Regular",
      "fontSize": 16,
      "fontWeight": "normal",
      "marginBottom": 20,
    }
  }
>
  Text of the main persuasion.
</Text>
`;

exports[`correctly renders the text elements 2`] = `
<Text
  style={
    Object {
      "fontFamily": "Quattrocento_700Bold",
      "fontSize": 33,
      "fontWeight": "700",
      "marginBottom": 20,
    }
  }
>
  Title!
</Text>
`;

exports[`correctly renders the text elements 3`] = `
<Text
  style={
    Object {
      "fontFamily": "Quattrocento_400Regular",
      "fontSize": 26,
      "fontWeight": "normal",
      "marginBottom": 20,
    }
  }
>
  I am a subtitle.
</Text>
`;

exports[`correctly renders the text elements 4`] = `
<Text
  style={
    Object {
      "fontFamily": "Quattrocento_400Regular",
      "fontSize": 20,
      "fontWeight": "normal",
      "marginBottom": 20,
    }
  }
>
  Not quite a subtitle, but still more important than the rest.
</Text>
`;

exports[`correctly renders the text elements 5`] = `
<Text
  style={
    Object {
      "fontFamily": "FanwoodText_400Regular_Italic",
      "fontSize": 12.5,
      "fontWeight": "normal",
      "marginBottom": 20,
    }
  }
>
  I am a Caption!
</Text>
`;
